camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -51,7 +52,8 @@
   /**
    * Construct using the json representation of the kafka request
    */
-  public KafkaReader(TaskAttemptContext context, EtlRequest request, int clientTimeout, int fetchBufferSize)
+  public KafkaReader(EtlInputFormat inputFormat, TaskAttemptContext context, EtlRequest request, 
```

Comment:
KafkaReader's old constructor already takes 4 arguments, adding one more doesn't seem a lot.  If you know how to get the InputFormat from context, then I don't need this extra argument.

For suggestion #2, I cannot use InputFormat, since this is a hadoop class which doesn't define method: createSimpleConsumer